core/src/bson_type_info.rs

```diff
@@ -19,6 +25,38 @@ pub struct BsonTypeInfo {
     pub is_auto_unique_value: Option<bool>,
     pub is_unsigned: Option<bool>,
     pub num_prec_radix: Option<u16>,
+    pub simple_type_info: Option<SimpleTypeInfo>,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct SimpleTypeInfo {
+    pub sql_type: SqlDataType,
+    pub non_concise_type: SqlDataType,
+    pub precision: Option<u16>,
+    pub octet_length: Option<u16>,
+    pub fixed_bytes_length: Option<u16>,
+}
+
+impl SimpleTypeInfo {
+    const fn new(precision: u16, octet_length: u16, fixed_bytes_length: u16) -> Option<Self> {
+        Some(Self {
+            sql_type: SqlDataType::VARCHAR,
+            non_concise_type: SqlDataType::VARCHAR,
+            precision: Some(precision),
+            octet_length: Some(octet_length),
+            fixed_bytes_length: Some(fixed_bytes_length),
+        })
+    }
+
+    const fn default() -> Option<Self> {
+        Some(Self {
+            sql_type: SqlDataType::VARCHAR,
+            non_concise_type: SqlDataType::VARCHAR,
+            precision: None,
+            octet_length: None,
+            fixed_bytes_length: None,
+        })
+    }
```

Comment:
Functions need to be constant to work in constant declarations, so I had to get rid of the `default()` trait. Instead, I just moved the `default()` function into the `impl SimpleTypeInfo`.